src/client/components/ModalNew.tsx

```diff
@@ -95,6 +95,18 @@ export const Modal = ({
 
   const hasOnCloseFunction = Boolean(onClose)
 
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'Escape' && !!onClose) {
+        onClose()
+      }
+    }
+    window.addEventListener('keydown', handleKeyDown)
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown)
+    }
+  }, [hasOnCloseFunction, onClose])
```

Comment:
hasOnCloseFunction isn't used in this hook.  Let's investigate why this isn't a lint error, it should be

Reply:
Yeah, didn't even think about it!